File: src/main/java/com/rednorte/sigle/pacientes_service/model/Notificacion.java
package com.rednorte.sigle.pacientes_service.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "notificaciones")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notificacion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "paciente_id")
    private Long pacienteId;
    
    private String tipo;
    private String canal;
    private String mensaje;
    private String estado;
    private Integer intentos;
    
    @Column(name = "enviado_en")
    private LocalDateTime enviadoEn;
    
    @Column(name = "evento_origen")
    private String eventoOrigen;
    
    @Column(name = "creado_en")
    private LocalDateTime creadoEn;

    @Column(name = "leido")
    @Builder.Default
    private Boolean leido = false;
    
    @PrePersist
    protected void onCreate() {
        creadoEn = LocalDateTime.now();
        if (intentos == null) intentos = 0;
        if (estado == null) estado = "PENDIENTE";
        if (leido == null) leido = false;
    }
}

File: src/models/Notificacion.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Notificacion = sequelize.define('Notificacion', {
    id: {
        type: DataTypes.BIGINT,
        autoIncrement: true,
        primaryKey: true,
    },
    pacienteId: {
        type: DataTypes.BIGINT,
        allowNull: false,
        field: 'paciente_id',
    },
    tipo: { type: DataTypes.STRING },
    canal: { type: DataTypes.STRING },
    mensaje: { type: DataTypes.TEXT },
    estado: { type: DataTypes.STRING, defaultValue: 'PENDIENTE' },
    intentos: { type: DataTypes.INTEGER, defaultValue: 0 },
    enviadoEn: { type: DataTypes.DATE, allowNull: true, field: 'enviado_en' },
    eventoOrigen: { type: DataTypes.STRING, allowNull: true, field: 'evento_origen' },
    creadoEn: { type: DataTypes.DATE, defaultValue: DataTypes.NOW, field: 'creado_en' },
}, {
    tableName: 'notificaciones',
    timestamps: false,
});

module.exports = Notificacion;